lib/src/container/store.rs

```diff
@@ -1125,7 +1114,22 @@ pub fn image_filtered_content_warning(
     Ok(r)
 }
 
-/// Remove the specified image references.
+/// Remove the specified image reference.  If the image is already
+/// not present, this function will successfully perform no operation.
+///
+/// This function assumes no transaction is active on the repository.
+/// The underlying layers are *not* pruned; that requires a separate invocation
+/// of [`gc_image_layers`].
+#[context("Pruning {img}")]
+pub fn remove_image(repo: &ostree::Repo, img: &ImageReference) -> Result<bool> {
+    let ostree_ref = &ref_for_image(img)?;
+    let found = repo.resolve_rev(ostree_ref, true)?.is_some();
+    repo.set_ref_immediate(None, ostree_ref, None, gio::NONE_CANCELLABLE)?;
```

Comment:
Nit: just curious, is there a reason why you decided to call this also in the not found case? I know you mentioned this is idempotent, but I guess we could just skip an additional trip through libostree in that case.

Reply:
Good point, adjusted!